snakebids/tests/test_app.py

```diff
@@ -33,6 +38,64 @@ def app(mocker: MockerFixture):
     return app
 
 
+class TestUpdateConfig:
+    @given(
+        st.one_of(
+            st.none(),
+            st.tuples(
+                sb_st.bids_entity(),
+                st.one_of(
+                    sb_st.bids_value(),
+                    st.booleans(),
+                    st.lists(sb_st.bids_value(), min_size=1),
+                ),
+            ),
+        ),
+        st.one_of(st.none(), st.lists(sb_st.bids_entity())),
+        st.one_of(
+            st.none(),
+            st.text(
+                alphabet=st.characters(
+                    blacklist_categories=("Cs",), blacklist_characters=("\x00",)
+                )
+            ),
+        ),
+    )
+    def test_magic_args(
+        self,
+        filters: tuple[BidsEntity, Iterable[str]] | None,
+        wildcards: Iterable[BidsEntity] | None,
+        path: str | None,
+    ):
+        config_copy = copy.deepcopy(config)
+        config_copy["bids_dir"] = "root"  # type: ignore
+        config_copy["output_dir"] = "app"  # type: ignore
+        args = SnakebidsArgs(
+            force=False,
+            outputdir=Path("app"),
+            snakemake_args=[],
+            args_dict={
+                "filter_bold": {filters[0].entity: filters[1]} if filters else None,
+                "wildcards_bold": [wildcard.entity for wildcard in wildcards]
+                if wildcards
+                else None,
+                "path_bold": path,
+            },
+        )
+        update_config(config_copy, args)
+        inputs_config: InputsConfig = cast(InputsConfig, config_copy["pybids_inputs"])
+        if filters:
+            assert (
+                inputs_config["bold"].get("filters", {}).get(filters[0].entity)
+                == filters[1]
+            )
+        if wildcards:
+            for entity in [wildcard.entity for wildcard in wildcards]:
```

Comment:
You can do a set subtraction here:
```suggestion
            assert not {wildcard.entity for wildcard in wildcards} - set(inputs_config["bold"].get("wildcards", []))
```
Alternatively a length comparison on the original vs an intersection

Reply:
Going from here, I landed on using the `<=` set operator

Reply:
Ah, cool. Never realized that was an option